py/miniconf-mqtt/miniconf/miniconf.py

```diff
@@ -39,8 +43,27 @@ def __init__(self, client, prefix):
         self.prefix = prefix
         self.inflight = {}
         self.client.on_message = self._handle_response
+        self.client.on_subscribe = self._handle_subscription
         self.response_topic = f'{prefix}/response'
-        self.client.subscribe(self.response_topic)
+        mid = self.client.subscribe(f'{prefix}/response')
+        self._pending_subscriptions = {mid: asyncio.get_running_loop().create_future()}
+
+
+    async def subscriptions_complete(self):
+        """ Wait for all pending subscriptions to complete. """
+        for subscription in list(self._pending_subscriptions.values()):
```

Comment:
This listification is required because we're modifying `self._pending_subscriptions` when we do `await`. The `list()` call forces a complete iteration and copy of the values in the dict before the first `await`

Reply:
Ah. Ack!